core/src/main/scala/org/apache/spark/sql/delta/GeneratedColumn.scala

```diff
@@ -161,6 +162,37 @@ object GeneratedColumn extends DeltaLogging with AnalysisHelper {
     spark.sessionState.sqlParser.parseExpression(exprString)
   }
 
+  /**
+   * SPARK-27561 added support for lateral column alias. This means generation expressions that
+   * reference other generated columns no longer fail analysis in `validateGeneratedColumns`.
+   *
+   * This method checks for and throws an error if:
+   * - A generated column references itself
+   * - A generated column references another generated column
+   */
+  def validateColumnReferences(
```

Comment:
is there test coverage for this?

Reply:
Yes in `GeneratedColumnSuite`, tests fail without this change.